### PR TITLE
Consider that arg keep_prob to dropout might be an int

### PR DESCRIPTION
**Pull request by gnebehay**
Calling

```
tf.nn.dropout(layer, dropout=0.0)
```

raises a `ValueError`, as expected. However, since the condition for the ValueError reads as

```
if isinstance(keep_prob, float) and not 0 < keep_prob <= 1:
    raise ValueError(...)
```

a call like this succeeds:

```
tf.nn.dropout(layer, dropout=0)
```

This problem occurs whenever an int is passed that is not 1 (the only allowed int value). This PR contains a simple workaround by considering that the argument might be an int.

### CODE REVIEW

**Comment by gnebehay** [comment]
I signed it!

File: tensorflow/python/ops/nn_ops.py [hunk 1]
```diff
@@ -1109,7 +1109,7 @@ def dropout(x, keep_prob, noise_shape=None, seed=None, name=None):
   """
   with ops.name_scope(name, "dropout", [x]) as name:
     x = ops.convert_to_tensor(x, name="x")
-    if isinstance(keep_prob, float) and not 0 < keep_prob <= 1:
+    if (isinstance(keep_prob, int) or isinstance(keep_prob, float)) and not 0 < keep_prob <= 1:
       raise ValueError("keep_prob must be a scalar tensor or a float in the "
```
Comment (rmlarsen) [inline tensorflow/python/ops/nn_ops.py]: Update error message?

**Comment by rmlarsen** [comment]
@tensorflow-jenkins test this please

File: tensorflow/python/ops/nn_ops.py (hunk 1 above)
Comment (mrry) [inline tensorflow/python/ops/nn_ops.py]: Sorry for the drive-by, but `isinstance(keep_prob, (float, int))` would be more idiomatic (and compact!) here.

Comment (gnebehay) [inline tensorflow/python/ops/nn_ops.py]: It seems that the problem prevails when long ints are passed, e.g. 0L.

If you can afford an additional import,

import numbers
isinstance(keep_prob, numbers.Real)

appears to be the proper solution.

http://stackoverflow.com/questions/4541155/check-if-a-number-is-int-or-float

Comment (rmlarsen) [inline tensorflow/python/ops/nn_ops.py]: SGTM, please go ahead and make that change.

**Comment by gnebehay** [comment]
Updated the commit to check if keep_prob is a real number.

**Comment by rmlarsen** [comment]
@tensorflow-jenkins test this please

**Comment by rmlarsen** [comment]
@gnebehay thanks for fixing this!
